snuba/admin/views.py

```diff
@@ -23,6 +30,41 @@ def health() -> Response:
     return Response("OK", 200)
 
 
+@application.route("/clickhouse_queries")
+def clickhouse_queries() -> Response:
+    res = [q.to_json() for q in SystemQuery.all_queries()]
+    return make_response(jsonify(res), 200)
+
+
+# Sample cURL command:
+#
+# curl -X POST \
+#  -d '{"query_name": "ActivePartitions"}' \
+#  -H 'Content-Type: application/json' \
+#  http://localhost:1219/run_clickhouse_system_query
+@application.route("/run_clickhouse_system_query", methods=["POST"])
+def clickhouse_system_query() -> Response:
+    req = request.get_json()
+    try:
+        results, columns = run_system_query_on_host_by_name(
+            req.get("host", "localhost"),
+            req.get("port", 9000),
+            req.get("storage", "transactions"),
+            req.get("query_name"),
+        )
+        res = {}
+        res["column_names"] = [name for name, _ in columns]
+        res["rows"] = []
+        for row in results:
+            res["rows"].append([str(col) for col in row])
```

Comment:
Might be nice to have a column formatter that keeps the original types intact where possible